optimum/exporters/executorch/integrations.py

```diff
@@ -24,15 +24,274 @@
     PreTrainedModel,
     StaticCache,
     T5ForConditionalGeneration,
+    VoxtralForConditionalGeneration,
     WhisperForConditionalGeneration,
 )
 from transformers.generation.configuration_utils import GenerationConfig
+from transformers.integrations.executorch import TorchExportableModuleForDecoderOnlyLM, sdpa_mask_without_vmap
+from transformers.masking_utils import AttentionMaskInterface
+from transformers.modeling_utils import AttentionInterface
 
 from optimum.executorch.attentions.custom_sdpa import get_custom_sdpa_for_ring_kv_cache
 
 from .utils import save_config_to_constant_methods
 
 
+class VoxtralEncoderExportableModule(torch.nn.Module):
+    """
+    Subgraph which handles all of the audio-related work: encoder, multimodal projection, combinining with text tokens.
+    The result of this subgraph should stream directly into the decoder subgraph.
+    """
+
+    def __init__(self, model: torch.nn.Module):
+        super().__init__()
+        self.audio_encoder = model.audio_tower
+        self.mm_projector = model.multi_modal_projector
+        self.intermediate_size = model.config.audio_config.intermediate_size
+        self.audio_token_id = model.config.audio_token_id
+        self.metadata = save_config_to_constant_methods(model.config.text_config, model.generation_config)
+
+    def prepare_export_inputs(self):
+        chunk_length = (
+            self.audio_encoder.config.max_source_positions
+            * self.audio_encoder.conv1.stride[0]
+            * self.audio_encoder.conv2.stride[0]
+        )
+        input_features = torch.rand(3, 128, chunk_length)  # (bsz, features, seq_len)
+
+        max_audio_len = 120  # In s, should be a multiple of 30. TODO: make configurable.
+        max_seq_len = self.metadata.get("get_max_seq_len")
+        dynamic_shapes = {
+            "input_features": {
+                0: torch.export.Dim("enc_batch_size_dim", min=1, max=max_audio_len // 30),
+            },
+        }
+
+        return input_features, dynamic_shapes
+
+    def forward(
+        self,
+        input_features: torch.FloatTensor,
+    ):
+        """
+        Forward pass of the Voxtral encoder module.
+
+        Args:
+            input_features (torch.FloatTensor): Raw audio features with shape (batch_size, features, seq_len).
+
+        Returns:
+            torch.FloatTensor: Combined embeddings with audio tokens replaced by audio embeddings.
+        """
+        audio_outputs = self.audio_encoder(input_features)
+        audio_hidden_states = audio_outputs.last_hidden_state
+        audio_hidden_states = audio_hidden_states.reshape(-1, self.intermediate_size)
+        audio_embeds = self.mm_projector(audio_hidden_states)
+        return audio_embeds
```

Comment:
So why do we need this https://github.com/huggingface/transformers/blob/main/src/transformers/models/voxtral/modeling_voxtral.py#L449-L453 here? Can the original hf model not be used directly?

Reply:
Yeah, just need to have that in a `forward`. Also don't know if we can rely on the function name `get_audio_embeds` across different audio models.